################################################################################
SHORT DESCRIPTION: 
################################################################################
Export private Linux interfaces for xenfb2 to be built out of tree.

################################################################################
LONG DESCRIPTION: 
################################################################################

################################################################################
CHANGELOG 
################################################################################
Original Author: unknown
Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>

################################################################################
REMOVAL 
################################################################################
When xenfb2 is not used anymore.

################################################################################
UPSTREAM PLAN
################################################################################
None this is specific to xenfb2 driver.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
xenfb2 will not build without set_phys_to_machine/zap_page_range exported in
Linux headers.

################################################################################
PATCHES 
################################################################################
Index: linux-3.18.16/arch/x86/xen/p2m.c
===================================================================
--- linux-3.18.16.orig/arch/x86/xen/p2m.c	2015-06-14 18:19:31.000000000 +0200
+++ linux-3.18.16/arch/x86/xen/p2m.c	2015-06-19 15:18:59.269681707 +0200
@@ -873,6 +873,7 @@
 
 	return true;
 }
+EXPORT_SYMBOL_GPL(set_phys_to_machine);
 
 #define M2P_OVERRIDE_HASH_SHIFT	10
 #define M2P_OVERRIDE_HASH	(1 << M2P_OVERRIDE_HASH_SHIFT)
Index: linux-3.18.16/mm/memory.c
===================================================================
--- linux-3.18.16.orig/mm/memory.c	2015-06-14 18:19:31.000000000 +0200
+++ linux-3.18.16/mm/memory.c	2015-06-19 15:18:59.439679723 +0200
@@ -1390,6 +1390,7 @@
 	mmu_notifier_invalidate_range_end(mm, start, end);
 	tlb_finish_mmu(&tlb, start, end);
 }
+EXPORT_SYMBOL_GPL(zap_page_range);
 
 /**
  * zap_page_range_single - remove user pages in a given range
Index: linux-3.18.16/drivers/video/fbdev/Kconfig
===================================================================
--- linux-3.18.16.orig/drivers/video/fbdev/Kconfig	2015-06-14 18:19:31.000000000 +0200
+++ linux-3.18.16/drivers/video/fbdev/Kconfig	2015-06-19 15:18:59.556345027 +0200
@@ -2274,6 +2274,16 @@
 	  frame buffer driver.  It communicates with a back-end
 	  in another domain.
 
+config XEN_FBDEV2_FRONTEND
+        tristate "Xen virtual frame buffer support for XenClient"
+        depends on FB && XEN
+        select FB_CFB_COPYAREA
+        select FB_CFB_IMAGEBLIT
+        select FB_CFB_FILLRECT
+        default y
+        help
+          Hack to compile the needed functions for xenfb2
+
 config FB_METRONOME
 	tristate "E-Ink Metronome/8track controller support"
 	depends on FB
